src/modules/previewpane/powerpreview/settings.cpp

```diff
@@ -0,0 +1,145 @@
+#include "pch.h"
+#include <common.h>
+#include "settings.h"
+#include "trace.h"
+
+namespace PowerPreviewSettings
+{
+	extern "C" IMAGE_DOS_HEADER __ImageBase;
+
+	// Base Settinngs Class Implementation
+	FileExplorerPreviewSettings::FileExplorerPreviewSettings(bool state, std::wstring name, std::wstring description)
+	{
+        this->m_isPreviewEnabled = std::move(state);
+		this->m_name = std::move(name);
+        this->m_description = std::move(description);
+	}
+
+	FileExplorerPreviewSettings::FileExplorerPreviewSettings(bool state)
+    {
+        this->m_isPreviewEnabled = std::move(state);
+    }
+
+	FileExplorerPreviewSettings::FileExplorerPreviewSettings()
+	{
+		this->m_isPreviewEnabled = false;
+		this->m_name = L"_UNDEFINED_";
+		this->m_description = L"_UNDEFINED_";
+	}
+
+	bool FileExplorerPreviewSettings::GetState() const
+	{
+		return this->m_isPreviewEnabled;
+	}
+
+	void FileExplorerPreviewSettings::SetState(bool state)
+	{
+        this->m_isPreviewEnabled = std::move(state);
+	}
+
+	void FileExplorerPreviewSettings::LoadState(PowerToysSettings::PowerToyValues &settings)
+	{
```

Comment:
Passing by reference is better here than by value (the way it was before).  const reference would be ideal, as that enforces that your not calling methods with side effects.  Do you still plan to update the PowerToysSettings to be const correct in a different PR?

Reply:
I thought of looking at it on another PR as I have another task dependent on this PR. Plus I will also need to look at the impact of this on other PowerToys that uses these methods.

Reply:
Totally on board with this!  :) It prevents the PR from growing.  Can you add a github issue/ADO task for this and assign it to yourself so that we can track it.  Please also assign this to yourself.